src/d/actor/d_a_tbox.cpp

```diff
@@ -202,23 +202,196 @@ daTbox_HIO_c::daTbox_HIO_c() {
+            return cPhs_ERROR_e;
+        }
+
+        modelFlags |= 0x1200;
+    }
+
+    // Load color anim, if requested
+    if (mdlInfo.brkId > 0) {
+        mpAppearRegAnm = new mDoExt_brkAnm();
+        if (mpAppearRegAnm == NULL) {
+            return cPhs_ERROR_e;
+        }
+
+        J3DAnmTevRegKey* appearRegData = (J3DAnmTevRegKey*)dComIfG_getObjectRes("Dalways", mdlInfo.brkId);
+        if (mpAppearRegAnm->init(modelData, appearRegData, true, 0, 1.0f, 0, -1, false, 0) == 0) {
+            return cPhs_ERROR_e;
+        }
+
+        modelFlags |= 0xF020000;
+    }
+
+    // Create model
+    mChestMdl = mDoExt_J3DModel__create(modelData, 0x80000, modelFlags);
+    if (mChestMdl == NULL) {
+        return cPhs_ERROR_e;
+    }
+
+    // Set up Triforce platform for the type that spawns via Wind Waker song
+    if (getFuncType() == FUNC_TYPE_TACT) {
+        modelData = (J3DModelData*)dComIfG_getObjectRes("Dalways", 0x17);
+        JUT_ASSERT(0xE2, modelData);
+
+        mTactPlatformMdl = mDoExt_J3DModel__create(modelData, 0x80000, 0x11000000);
+        if (mTactPlatformMdl == NULL) {
+            return cPhs_ERROR_e;
+        }
+
+        J3DAnmTevRegKey* tactPlatformBrk = (J3DAnmTevRegKey*)dComIfG_getObjectRes("Dalways", 0x1F);
+        if (mTactPlatformBrk.init(modelData, tactPlatformBrk, true, 0, 1.0f, 0, -1, false, 0) == 0) {
+            return cPhs_ERROR_e;
+        }
+    }
+
+    mChestMdl->setBaseScale(mScale);
+
+    mDoMtx_stack_c::transS(current.pos);
+    mDoMtx_stack_c::YrotM(current.angle.y);
+    mChestMdl->setBaseTRMtx(mDoMtx_stack_c::get());
+
+    if (getFuncType() == FUNC_TYPE_TACT) {
+        mDoMtx_stack_c::transM(0.0f, 1.0f, 0.0f);
+        mTactPlatformMdl->setBaseTRMtx(mDoMtx_stack_c::get());
+    }
+
+    mDoMtx_copy(mDoMtx_stack_c::get(), mMtx);
+
+    return cPhs_COMPLEATE_e;
 }
 
 /* 00000598-00000764       .text effectShapeSet__8daTbox_cFv */
-void daTbox_c::effectShapeSet() {
-    /* Nonmatching */
+s32 daTbox_c::effectShapeSet() {
+    J3DModelData* flashModelData = (J3DModelData*)dComIfG_getObjectRes("Dalways", 0x16);
+    JUT_ASSERT(0x117, flashModelData != 0);
+    
+    mpFlashMdl = mDoExt_J3DModel__create(flashModelData, 0x80000, 0x1000200);
+    if (mpFlashMdl == NULL) {
+        return cPhs_ERROR_e;
+    }
+
+    J3DAnmTransform* flashAnm = (J3DAnmTransform*)dComIfG_getObjectRes("Dalways", 0x0B);
+    if (mFlashAnm.init(flashModelData, flashAnm, true, 0, 1.0f, 0, -1, false) == 0) {
+        return cPhs_ERROR_e;
+    }
+
+    J3DAnmTextureSRTKey* flashTexAnm = (J3DAnmTextureSRTKey*)dComIfG_getObjectRes("Dalways", 0x25);
+    if (mFlashTexAnm.init(flashModelData, flashTexAnm, true, 0, 1.0f, 0, -1, false, 0) == 0) {
+        return cPhs_ERROR_e;
+    }
+
+    J3DAnmTevRegKey* flashRegAnm = (J3DAnmTevRegKey*)dComIfG_getObjectRes("Dalways", 0x1E);
+
+    s32 regInit = mFlashRegAnm.init(flashModelData, flashRegAnm, true, 0, 1.0f, 0, -1, false, 0);
+    return regInit != 0 ? 4 : 5;
```

Comment:
is this not `regInit != 0 ? cPhs_COMPLEATE_e : cPhs_ERROR_e;`? mind putting a comment if not